activerecord/test/cases/adapters/abstract_mysql_adapter/connection_test.rb

```diff
@@ -216,6 +216,43 @@ def test_release_non_existent_advisory_lock
       "expected release_advisory_lock to return false when there was no lock to release"
   end
 
+  def test_version_string
+    @connection.stub(:get_full_version, "8.0.35-0ubuntu0.22.04.1") do
+      assert_equal "8.0.35", @connection.get_database_version.to_s
+    end
+
+    @connection.stub(:get_full_version, "5.7.0") do
+      assert_equal "5.7.0", @connection.get_database_version.to_s
+    end
+  ensure
+    @connection.pool.pool_config.server_version = nil
+  end
+
+  def test_version_string_invalid
+    @connection.stub(:get_full_version, "some-database-proxy") do
+      error = assert_raises(ActiveRecord::ActiveRecordError) do
+        @connection.get_database_version
+      end
+      assert_match(/Unable to parse MySQL version/, error.message)
```

Comment:
A nitpick really but perhaps we could assert the full error message to explicitly show that the `full_version_string` is included in the message.

As I was about to suggest including `full_version_string` into the message based on this assertion until I realized that we just don't assert for it 🙈